File: clients/javascript/src/models/getStreamOut.ts
// this file is @generated
import {
    type StorageType,
    StorageTypeSerializer,
} from './storageType';





export interface GetStreamOut {
    createdAt: Date;
maxByteSize?: number | null;
name: string;
retentionPeriodSeconds?: number | null;
storageType: StorageType;
updatedAt: Date;
}

export const GetStreamOutSerializer = {
    _fromJsonObject(object: any): GetStreamOut {
        return {
            createdAt: new Date(object['created_at']),
            maxByteSize: object['max_byte_size'],
            name: object['name'],
            retentionPeriodSeconds: object['retention_period_seconds'],
            storageType: StorageTypeSerializer._fromJsonObject(object['storage_type']),
            updatedAt: new Date(object['updated_at']),
            };
    },

    _toJsonObject(self: GetStreamOut): any {
        return {
            'created_at': self.createdAt,
            'max_byte_size': self.maxByteSize,
            'name': self.name,
            'retention_period_seconds': self.retentionPeriodSeconds,
            'storage_type': StorageTypeSerializer._toJsonObject(self.storageType),
            'updated_at': self.updatedAt,
            };
    }
}

File: clients/rust/src/models/get_stream_out.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::storage_type::StorageType;

#[non_exhaustive]
#[derive(Clone, Debug, Deserialize, Serialize)]
pub struct GetStreamOut {
    pub created_at: jiff::Timestamp,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub max_byte_size: Option<u64>,

    pub name: String,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub retention_period_seconds: Option<u64>,

    pub storage_type: StorageType,

    pub updated_at: jiff::Timestamp,
}

impl GetStreamOut {
    pub fn new(
        created_at: jiff::Timestamp,
        name: String,
        storage_type: StorageType,
        updated_at: jiff::Timestamp,
    ) -> Self {
        Self {
            created_at,
            max_byte_size: None,
            name,
            retention_period_seconds: None,
            storage_type,
            updated_at,
        }
    }
}
